### PR TITLE
Temp Regridded - fix #699

### LINKED ISSUE #699: anmn_temp_gridded_product.py - Averaging process
I anticipate the current averaging process being an issue for users.

Ideally, one would like to retrieve averaged values that can fall on minute 00, regardless of when the deployment starts.

So, if a deployment starts at 10:06 and we want to perform 30min averages, then our first averaging cell should have its middle falling on 10:00, hence starts at 9:45 and finishes at 10:15. It is likely that for this first cell, there is not enough samples to meet the minimum required in the algorithm, but this doesn't matter.

### PR DESCRIPTION
**Pull request by lbesnard**
@ggalibert  https://github.com/aodn/data-services/issues/699

### CODE REVIEW

File: ANMN/temperature_grid_product/anmn_temp_gridded_product.py [hunk 1]
```diff
@@ -512,8 +529,8 @@ def main(incoming_file_path, deployment_code, output_dir, plot_comparaison=False
     global fv01_dir
     s3_bucket_prefix = 'https://s3-ap-southeast-2.amazonaws.com/imos-data'
     logging           = IMOSLogging()
-    logger            = logging.logging_start(os.path.join(os.environ['WIP_DIR'], 'anmn_temp_grid.log'))
-    list_fv01_url     = wfs_request_matching_file_pattern('anmn_ts_timeseries_map', '%%_FV01_%s%%' % deployment_code, s3_bucket_url=True)
+    logger            = logging.logging_start(os.path.join(output_dir, 'anmn_temp_grid.log'))
+    list_fv01_url     = wfs_request_matching_file_pattern('anmn_ts_timeseries_map', '%%_FV01_%s%%' % deployment_code, s3_bucket_url=True, url_column='file_url')
```
Comment (ggalibert) [inline ANMN/temperature_grid_product/anmn_temp_gridded_product.py]: These two changes above should go in a separate commit / PR.

**Comment by ggalibert** [comment]
We need a different logic since this is only going to work for a dataset that needs averaging over 30min, however the averaging time period is dependant on the site sampling regime (see [`get_frequency_step_in_deployment`](https://github.com/aodn/data-services/blob/98ebf9631889903561af3b47f0ec046e6f2e6f92/ANMN/temperature_grid_product/anmn_temp_gridded_product.py#L198)).

**Comment by lbesnard** [comment]
ok will think about this tomorrow

**Comment by lbesnard** [comment]
@ggalibert done some mods. I removed the 90' option. I think  using 120'  average could be a bit annoying.

File: ANMN/temperature_grid_product/anmn_temp_gridded_product.py [hunk 2]
```diff
@@ -172,11 +172,39 @@ def _perdelta(start, end, delta):
...
+    elif delta_in_minutes == 60:
+        if time_start.minute > 30:
+            time_start = time_start.replace(minute=0, second=0, microsecond=0) + timedelta(hours=1)
+        elif time_start.minte < 30:
+            time_start = time_start.replace(minute=0, second=0, microsecond=0)
+
+        if time_end.minute > 30:
+            time_end = time_end.replace(minute=0, second=0, microsecond=0) + timedelta(hours=1)
+        elif time_end.minute < 30:
+            time_end = time_end.replace(minute=0, second=0, microsecond=0)
```
Comment (ggalibert) [inline ANMN/temperature_grid_product/anmn_temp_gridded_product.py]: There is a more generic way you can do this for any delta_in_minutes value. Your new time_start for example could be set by something like:
```Matlab
time_start_in_minutes = floor(time_start_in_minutes/delta_in_minutes)*delta_in_minutes
```

File: ANMN/temperature_grid_product/anmn_temp_gridded_product.py [hunk 3]
```diff
@@ -229,8 +257,8 @@ def _sample_inter(sample_interval):
             average_window = 60
             n_val_step     = 3
         elif round(sample_interval) >= 1200:
-            average_window = 90
-            n_val_step     = 3
+            average_window = 60
+            n_val_step     = 2
```
Comment (ggalibert) [inline ANMN/temperature_grid_product/anmn_temp_gridded_product.py]: Why 2?

Comment (lbesnard) [reply ANMN/temperature_grid_product/anmn_temp_gridded_product.py]: it's the amount of valid measures (good qc) we need in the average window to perform a average. see with @bpasquer

**Comment by ggalibert** [comment]
Closing this since the following PR https://github.com/aodn/data-services/pull/809 includes some of the above modifications + a fix to https://github.com/aodn/data-services/issues/699 with the following commit: https://github.com/aodn/data-services/pull/809/commits/0e0cf84fb3a469d2ef540fdd2edfcd82bc8052e4.
